File: ToDoApp_backend/ViewModel/Schedule/ScheduleViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace ToDoApp_backend.ViewModel.Schedule
{
    public class ScheduleViewModel
    {
        public string ScheduleName { get; set; }
        public string Summary { get; set; }
        public DateTime StartDay { get; set; }
        public DateTime EndDay { get; set; }
        public int UsageTime { get; set; }
        public bool NoticeSettingFlg { get; set; }
        public bool LineFlg { get; set; }
        public bool SlackFlg { get; set; }
        public string LineAccount { get; set; }
        public string LinePassword { get; set; }
        public string SlackAccount { get; set; }
        public string SlackPassword { get; set; }
        public bool CategoryMasterFlg { get; set; }
        public long CategoryMasterId { get; set; }
    }
}

File: frontend_App/todo-list/types/ViewModel/ScheduleViewModel.ts
export interface IScheduleViewModel {
  ScheduleName: string // スケジュール名
  Summary: string // 概要説明
  StartDay: Date // 開始日
  EndDay: Date // 終了日
  UsageTime: number // 使用時間
  NoticeSettingFlg: boolean // 通知設定を表示するかどうか
  LineFlg: boolean // LINE設定
  LineAccount: string // LINEアカウント
  LinePassword: string // LINEパスワード
  SlackFlg: boolean // SLACK設定
  SlackAccount: string // SLACKアカウント
  SlackPassword: string // SLACKパスワード
  CategoryMasterFlg: boolean// カテゴリーマスタを表示するかどうか
  CategoryMasterId: number
}

export class ScheduleViewModel implements IScheduleViewModel {
  ScheduleName = ''
  Summary = ''
  StartDay = new Date()
  EndDay = new Date()
  UsageTime = 0
  NoticeSettingFlg = false
  LineFlg = false
  SlackFlg = false
  LineAccount = ''
  LinePassword = ''
  SlackAccount = ''
  SlackPassword = ''
  CategoryMasterFlg = false
  CategoryMasterId = 0
}
